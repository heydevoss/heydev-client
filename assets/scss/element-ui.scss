// Import Custom Variables
@import "variables";

/* icon font path, required */
$--font-path: '~element-ui/lib/theme-chalk/fonts';

@import "~element-ui/packages/theme-chalk/src/index";