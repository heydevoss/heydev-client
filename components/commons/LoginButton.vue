<template>
  <div>
    <a href="http://localhost:5000/auth/login" class="button--grey">Login</a>
  </div>
</template>

<script>
export default {
  name: 'LoginButton'
}
</script>
<style lang="scss" scoped></style>
