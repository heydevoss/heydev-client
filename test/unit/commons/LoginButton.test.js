import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import LoginButton from '@/components/commons/LoginButton.vue'
import { Button } from 'element-ui'

Vue.use(Button)

describe('LoginButton', () => {
  const wrapper = shallowMount(LoginButton, {
    stubs: {
      'el-button': Button
    }
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain button', () => {
    expect(wrapper.html()).toContain('Login')
    expect(wrapper.html().includes('<button>'))
  })

  it('should redirect when clicked', () => {
    const button = wrapper.find('button')
    button.trigger('click')
    expect(window.location.href).toBe('http://localhost/')
  })
})
