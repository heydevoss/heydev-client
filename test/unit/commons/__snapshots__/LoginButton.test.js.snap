// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton is a Vue instance 1`] = `
<div>
  <button
    class="el-button el-button--primary"
    type="button"
  >
    <!---->
    <!---->
    <span>
      Login
    </span>
  </button>
</div>
`;
